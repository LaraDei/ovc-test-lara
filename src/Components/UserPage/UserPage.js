import React, { Component } from 'react';
import { connect } from 'react-redux'
import SearchBar from '../SearchBar/SearchBar';
import TableStyle from '../TableStyle/TableStyle';
import {getUsers} from '../../Store/usersSlice';



class UserPage extends Component{

  componentDidMount(){
    this.props.getUsers();
  };

  render(){
    const usersData = this.props.users.users.map(user => {
      return {
        userId: user.id,
        name: user.name,
        email: user.email,
        city: user.address.city,
        company: user.company.name
      }
    });
    const headerList = ['Name','Email','City','Company'];
    return(
        <div className="user-page">
          <SearchBar/>
          <h1>List of Users</h1>
          <TableStyle columns={headerList} data={usersData}/>
        </div>
    )
    }
}
const mapStateToProps = (state) => {
  return {
    users : state.users
  }
};
const mapDispatchToProps = () => {
  return {
    getUsers
  }
};
export default connect(mapStateToProps, mapDispatchToProps())(UserPage);

