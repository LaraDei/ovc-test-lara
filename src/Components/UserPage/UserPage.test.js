import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import UserPage from "./UserPage";

describe("UserPage component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<UserPage />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders by default", () => {
    const wrapper = renderer.create(<UserPage />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it("filters array", () => {});
});
