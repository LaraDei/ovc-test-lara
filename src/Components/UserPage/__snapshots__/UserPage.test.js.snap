// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPage component renders by default 1`] = `
<div
  className="user-page"
>
  <div
    className="user-search-wrapper"
  >
    <h3>
      Search Users By Name:
    </h3>
    <div
      className="search-container"
    >
      <form>
        <input
          id="searchInput"
          name="search"
          onChange={[Function]}
          placeholder="Search.."
          type="text"
        />
        <button
          onClick={[Function]}
          type="clear"
        >
          Clear
        </button>
      </form>
    </div>
  </div>
  <h2>
    List of Users
  </h2>
  <div
    className="table-container"
  >
    <table
      className="table"
    >
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Email
          </th>
          <th>
            City
          </th>
          <th>
            Company
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
